service/keychain/cri/v1/cri.go

```diff
@@ -81,8 +81,7 @@ type instrumentedService struct {
 	configMu sync.Mutex
 }
 
-func (in *instrumentedService) credentials(imgRefSpec reference.Spec) (string, string, error) {
```

Comment:
> so that when we attempt to get credentials through our CRI implementation for a mirror, we try to use the credentials for the base host first

Sorry, this isn't really that clear. The CRI implementation doesn't blindly send the base host credentials over when it encounters a mirror. It still has to pass the [check in `ParseAuth`](https://github.com/awslabs/soci-snapshotter/blob/88bcf75788de454dd03137b65ed29c7e662e0333/service/resolver/cri.go#L369-L377) that ensures that the host provided by the authorizer matches the host of the base image ref. If that check fails we will not send the credentials over. 

eg: 

CRI sends us the creds for `registry.com/namespace/repo:latest`->

SOCI uses a configured mirror `mirror.com`->

We get a 401 from `mirror.com` so we call the creds function->

We initially closed over `registry.com/namespace/repo:latest` so we call the `ParseAuth` function passing in the `AuthConfig` for  `registry.com/namespace/repo:latest` and the host that the authorizer passed us `mirror.com`->

Assuming the `ServerAdress` field in the `AuthConfig` was populated, the check fails (`registry.com` != `mirror.com`) and we return an empty set of credentials

Reply:
I may have misinterpreted your question. We still get the host through because `multiCredFuncs` returns a normal `docker.Authorizer` compliant credential function that takes only a host. 

https://github.com/awslabs/soci-snapshotter/blob/af695d3a1b8efa087c9342811e253a7c7246ece8/service/resolver/registry.go#L186